library/core/src/sync/atomic.rs

```diff
@@ -24,26 +24,44 @@
 //!
 //! ## Memory model for atomic accesses
 //!
-//! Rust atomics currently follow the same rules as [C++20 atomics][cpp], specifically `atomic_ref`.
-//! Basically, creating a *shared reference* to one of the Rust atomic types corresponds to creating
-//! an `atomic_ref` in C++; the `atomic_ref` is destroyed when the lifetime of the shared reference
-//! ends. A Rust atomic type that is exclusively owned or behind a mutable reference does *not*
-//! correspond to an “atomic object” in C++, since the underlying primitive can be mutably accessed,
-//! for example with `get_mut`, to perform non-atomic operations.
+//! Rust atomics currently follow the same rules as [C++20 atomics][cpp], specifically the rules
+//! from the [`intro.races`][cpp-intro.races] section, without the "consume" memory ordering. Since
+//! C++ uses an object-based memory model whereas Rust is access-based, a bit of translation work
+//! has to be done to apply the C++ rules to Rust: whenever C++ talks about "the value of an
+//! object", we understand that to mean the resulting bytes obtained when doing a read. When the C++
+//! standard talks about "the value of an atomic object", this refers to the result of doing an
+//! atomic load (via the operations provided in this module). A "modification of an atomic object"
+//! refers to an atomic store.
+//!
+//! The most important aspect of this model is that conflicting non-synchronized accesses are
+//! Undefined Behavior unless both accesses are atomic. Here, accesses are *conflicting* if they
```

Comment:
No strong opinion, but note that `Undefined Behavior` has >900 hits in the rustc sources, over 60 of which are in `library`. `undefined behavior` occurs around 2x to 4x that much, so it is clearly in the majority, but it's not a clear consensus either.

Reply:
Yes, I almost mentioned legal writing as an exception to what's otherwise common in English.  But if we're going to do that, then we should capitalize Data Races and many other terms.

Reply:
> The reference also has a few occurrences of `Undefined Behavior`.

Just checked.  These were all added by you :)

Reply:
Guilty as charged, I guess. ;)

Reply:
A fair chunk of the Rust community uses capitalization like The Book when saying things like "Are you sure that isn't in The Book?" and referring specifically to https://doc.rust-lang.org/book/ not some previously-mentioned book. I think that usage style is primarily drawn from religious backgrounds: https://www.englishgrammar101.com/module-11/capitalization/lesson-4/names-of-groups-of-people
> Believers also capitalize pronouns (like he and him) when referring to God. 

There are other more vague ways people will use caps for emphasis, for example I might say "That sounds to me like A Bad Idea". In this case the caps don't change the meaning but they provide a sort of emphasis.

Not that I think either of those are immediately relevant to this text. But younger English speakers/writers use caps in the above ways, because the language is evolving.